test/datasource.test.js

```diff
@@ -129,7 +129,55 @@ describe('loopback:datasource generator', function() {
     });
   });
 
-  it('should support IBM Object Storage ', function(done) {
+  it('should support MongoDB', function(done) {
+    var datasourceGen = givenDataSourceGenerator();
+    helpers.mockPrompt(datasourceGen, {
+      name: 'mongodb-datasource',
+      connector: 'mongodb',
+      installConnector: false,
+    });
+
+    datasourceGen.run(function() {
+      var datasources = readDataSourcesJsonSync('server');
+      // eslint-disable-next-line no-unused-expressions
+      expect(datasources['mongodb-datasource']).to.exist;
+      var ds = datasources['mongodb-datasource'];
+      expect(ds.name).to.equal('mongodb-datasource');
+      expect(ds.connector).to.equal('mongodb');
+      var pkg = fs.readFileSync(
+        path.join(SANDBOX, 'package.json'), 'UTF-8');
+      pkg = JSON.parse(pkg);
```

Comment:
Please consider extracting these tree lines into a shared helper function, e.g. `readPackageJson`, similar to the existing `readDataSourcesJsonSync`.